cpg_infra/abstraction/gcp.py

```diff
@@ -492,14 +493,22 @@ def get_credentials_for_machine_account(self, resource_key, account):
 
     def create_group(self, name: str) -> Any:
         mail = f'{name}@{self.config.gcp.groups_domain}'
-        return gcp.cloudidentity.Group(
+        group = gcp.cloudidentity.Group(
             self.get_pulumi_name(name + '-group'),
             display_name=name,
             group_key=gcp.cloudidentity.GroupGroupKeyArgs(id=mail),
             labels={'cloudidentity.googleapis.com/groups.discussion_forum': ''},
             parent=f'customers/{self.config.gcp.customer_id}',
             opts=pulumi.resource.ResourceOptions(depends_on=[self._svc_cloudidentity]),
         )
+        # Allow domain-external members in the group.
+        GroupSettings(
+            f'{name}-group-settings',
```

Comment:
Could be worth calling the `get-pulumi-name` to ensure the ID is definitely unique, but adding `gcp-` to the start should also be fine.

```suggestion
            f'gcp-{name}-group-settings',
```